homeassistant/components/vesync/coordinator.py

```diff
@@ -0,0 +1,42 @@
+"""Representation of VeSync data coordinator."""
+
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+from typing import Any
+
+from pyvesync import VeSync
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import UPDATE_INTERVAL
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class VeSyncDataCoordinator(DataUpdateCoordinator):
```

Comment:
```suggestion
class VeSyncDataCoordinator(DataUpdateCoordinator[..]):
```
Add the return type of `_async_update_data` to the type slot of the coordinator